### ISSUE TITLE
Desired Development: Daemonize Actors

### ISSUE THREAD

**Issue by henricasanova** [wrench-project/wrench#8]
S4U does/will provide a way to daemonize actors, which may be used to simply a bit the WRENCH code? For instance, the Alarm services?

**Comment by henricasanova** [comment]
As we were implementing the notion that there can be multiple network proximity services, and thus modifying the Terminator implementation, it became clear that this whole "all services must be shutdown so that the simulation doen't end with the 'dealock or not perfectly clean' message" idea was problematic.  If 2 WMS's use the same network proximity service, then the 1st WMS to complete shouldn't shut it down, but the 2nd should. Yet, the network proximity service isn't passed to the WMS constructor, and so it's impossible to make that call.  

So, once more, I tried, in the "daemonize" branch to make services "daamonized" (i.e., SimGrid will not complained about hanging actors if these actors are daemonized).  And, finaly, it worked! (I guess S4U was fixed.... last time I tried was perhaps 4 months ago). 

So, what I propose is: 
  - Remove the whole Terminator thing (sorry Rafael, it was nice while it lasted)
  - Never have a WMS call some "shutdown all" function (because it's not clear what "all" means)
     (However a WMS can of course call Service::stop() for whatever service if it so desires)
  - Have ALL services/actors/helpers/managers daemonized
  - Only the WMS actors are not daemonized

The result is: the simulation terminates when all WMS have returned, and  not complain if other actors are still running.  This is just like the real world (since a  completing WMS does not, for instance, shutdown the file registry catalog!)

I've implemented all the above in the "daemonize" branch, and all tests pass on my Mac and on my Linux box. So, I am ready to merge this into master and never look back.

Rafael, since you implemented the Terminator, let me know whether you see an issue with this.
